production/ksonnet/loki/boltdb_shipper.libsonnet

```diff
@@ -70,6 +70,8 @@
     k.util.configVolumeMount('loki', '/etc/loki/config') +
     k.util.configVolumeMount('overrides', '/etc/loki/overrides') +
     statefulSet.mixin.spec.updateStrategy.withType('RollingUpdate') +
-    statefulSet.mixin.spec.template.spec.securityContext.withFsGroup(10001)  // 10001 is the group ID assigned to Loki in the Dockerfile
+    statefulSet.mixin.spec.template.spec.securityContext.withFsGroup(10001) +  // 10001 is the group ID assigned to Loki in the Dockerfile
+    statefulSet.mixin.spec.strategy.rollingUpdate.withMaxSurge(0) +
+    statefulSet.mixin.spec.strategy.rollingUpdate.withMaxUnavailable(1)
```

Comment:
maxSurge and maxUnavailable are not supported by statefulset. There is an open [proposal to add support for maxUnavailable to statefulsets](https://github.com/kubernetes/kubernetes/issues/68397) and maxSurge would not work with statefulset as per [comment](https://github.com/kubernetes/kubernetes/issues/68397#issuecomment-426007032) which makes sense.

Reply:
good catch @sandeepsukhani . Fixed :+1: